Add language toggle and store state in url

jwwood3 opened the pull request.

jwwood3 committed:
Accommodate smaller screens better

Accommodate smaller screens better

jwwood3 committed:
Start making text more localizable

Set up LANG object in model to store text in various languages. Replaced text in model, view, viewmodel, and index with references to LANG object. Still need to figure out text with variables in the middle, text on the maps, and text in the databases.

jwwood3 committed:
Start adding an options dropdown for language switching

jwwood3 committed:
Finish adding language toggle

currently reloads the page with a new active language using the query string

jwwood3 committed:
Store map data in/load from url query string

jwwood3 committed:
Finish storing data searches into url query string

I added an event to a dummy element to  trigger once variables are loaded. We have to wait until this happens before we can start loading any data specified by the url.

jwwood3 committed:
Reformat code

jwwood3 committed:
Remove some unnecessary changes

jwwood3 committed:
Update model.js

kisaacs commented:
I apologize for taking so long to get to this. There seems to be an issue when I hover over the map when there's data. The `this.model` is undefined in the `_update` function, used on line 499.

jwwood3 committed:
Fix reference error

The _update function returns a function, so "this" inside the function refers to the infoBox, not the viewModel itself, so the model needs to be referenced as this.m, not this.model

jwwood3 commented:
Thank you, I think this should fix it. Because that function is called by the infoBox, not the viewModel, it should be .m, not .model

kisaacs commented:
Thanks, looks good!
